src/netbase.cpp

```diff
@@ -77,15 +79,21 @@ bool static LookupIntern(const char *pszName, std::vector<CNetAddr>& vIP, unsign
     struct addrinfo aiHint;
     memset(&aiHint, 0, sizeof(struct addrinfo));
 
+    // We want a TCP port, which is a streaming socket type
     aiHint.ai_socktype = SOCK_STREAM;
     aiHint.ai_protocol = IPPROTO_TCP;
+    // We don't care which address family (IPv4 or IPv6) is returned
     aiHint.ai_family = AF_UNSPEC;
+    // If we allow lengthy lookups of hostnames, only return addresses whose
```

Comment:
What privacy concerns are there? What call-sites are you talking about?

Reply:
DNS lookups leak your query to your DNS server (or when done over tor, to the exit node that does the resolving).

Reply:
@sipa If I'm understanding correctly, one such instance is the "Resolve" stage of `ConnectNode` where if we have a name proxy, `Lookup` won't do external queries?

Reply:
If you set AI_NUMERICHOST then all `getaddrinfo` does is parse the IP address, and return the result immediately without network interaction.

Reply:
Ah, I thought that was obvious and wanted to explain why we needed `AI_ADDRCONFIG` for the case where `fAllowLookup` was true instead of just giving no flags. I should probably talk about both though.

Reply:
Updated: https://github.com/bitcoin/bitcoin/pull/15824/files#diff-4942713bcd2752bb531bf76c9a0ba38eR91